File: domain/src/main/java/com/github/dkurata38/liblio/domain/librarian/Librarian.java
package com.github.dkurata38.liblio.domain.librarian;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NonNull;
import lombok.ToString;

import com.github.dkurata38.liblio.domain.library.LibraryId;

@EqualsAndHashCode(of = {"libraryId"})
@ToString
public class Librarian {
    @Getter
    private final LibrarianId librarianId;
    @Getter
    private final LibraryId libraryId;
    @Getter
    private final Username username;
    @Getter
    private final Password password;

    public Librarian(@NonNull LibrarianId librarianId, @NonNull LibraryId libraryId,
                     @NonNull Username username, @NonNull Password password) {
        this.librarianId = librarianId;
        this.libraryId = libraryId;
        this.username = username;
        this.password = password;
    }
}

File: domain/src/main/kotlin/com/github/dkurata38/liblio/domain/librarian/Librarian.kt
package com.github.dkurata38.liblio.domain.librarian

import com.github.dkurata38.liblio.domain.library.LibraryId

open class Librarian(
    val librarianId: LibrarianId,
    val libraryId: LibraryId,
    val username: Username,
    val password: Password
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Librarian

        if (librarianId != other.librarianId) return false

        return true
    }

    override fun hashCode(): Int {
        return librarianId.hashCode()
    }

    override fun toString(): String {
        return "Librarian(librarianId=$librarianId, libraryId=$libraryId, username=$username, password=$password)"
    }
}
